.github/workflows/link-check.yaml

```diff
@@ -0,0 +1,31 @@
+name: external-link-check
+
+on: [pull_request]
+
+jobs:
+  base:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      # NOTE:
+      #  - the check process can be very fast,
+      #    but if `-c` is too large then `Too Many Requests (HTTP error 429)` may be reported from `https://github.com/*`.
+      #  - we hardcode `--document-root` to `/github/workspace` in the container now.
+      #  - we use `http://172.*` as sample addresses in some docs, so we need to exclude them.
+      - name: Check external links
+        id: lc
+        uses: peter-evans/link-checker@v1
+        with:
+          args: -c 32 -d /github/workspace -r -x "http://172.*|https://github.com/.*/pull/.*|https://github.com/.*/issues/.*|https://github.com/[^.\n]*$|http://127.0.0.1.*|http://localhost.*|http://.*:.*|https://static.googleusercontent.com/media/research.google.com/.*|https://www.crunchbase.com/organization/.*|http://www.brendangregg.com.*|https://blog.csdn.net.*|http://mysql-cocoa.sourceforge.net.*|https://tools.ietf.org.*|http://code.google.com.*|https://spark.apache.org.*|https://www.caasdata.com/|https://www.linkedin.com/company/.*|https://sea-group.org/.*|https://octave.sourceforge.io/database/index.html|https://www.teld.cn/|http://gnade.sourceforge.net/|https://en.wikipedia.org/wiki/WeBank_\(China|https://twitter.com/PingCAP|.*.md" .
```

Comment:
Why are we excluding the following sites?

> https://www.crunchbase.com/organization/.*|http://www.brendangregg.com.*|https://blog.csdn.net.*|http://mysql-cocoa.sourceforge.net.*|https://tools.ietf.org.*|http://code.google.com.*|https://spark.apache.org.*|https://www.caasdata.com/|https://www.linkedin.com/company/.*|https://sea-group.org/.*|https://octave.sourceforge.io/database/index.html|https://www.teld.cn/|http://gnade.sourceforge.net/|

Reply:
It's been tested in other repos that these links cause false alarms or links we want to ignore, so I keep them for this repo.

If you find some links that should not be excluded, I can remove them.

Reply:
Got it, thanks.